realm/realm-library/src/main/java/io/realm/internal/async/RealmThreadPoolExecutor.java

```diff
@@ -16,25 +16,29 @@
 
 package io.realm.internal.async;
 
+import java.io.File;
+import java.io.FileFilter;
 import java.util.concurrent.ArrayBlockingQueue;
 import java.util.concurrent.Callable;
 import java.util.concurrent.Future;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.locks.Condition;
 import java.util.concurrent.locks.ReentrantLock;
+import java.util.regex.Pattern;
 
-import io.realm.Realm;
+import edu.umd.cs.findbugs.annotations.SuppressWarnings;
 
 /**
  * Custom thread pool settings, instances of this executor can be paused, and resumed, this will also set
  * appropriate number of Threads & wrap submitted tasks to set the thread priority according to
  * <a href="https://developer.android.com/training/multiple-threads/define-runnable.html"> Androids recommendation</a>.
  */
 public class RealmThreadPoolExecutor extends ThreadPoolExecutor {
-    // reduce context switch by using a number of thread proportionate to the number of cores
-    // from AOSP https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/core/java/android/os/AsyncTask.java#182
-    private static final int CORE_POOL_SIZE = Runtime.getRuntime().availableProcessors() * 2 + 1;
+    private static final String SYS_CPU_DIR = "/sys/devices/system/cpu/";
```

Comment:
Sysfs as been part of Linux since 2.5.44, and Android Froyo was based on 2.6.32. A vendor might add custom SELinux rules to prevent access to `/sysfs`.